sycl/include/syclcompat/device.hpp

```diff
@@ -295,19 +438,18 @@ class device_ext : public sycl::device {
     return prop;
   }
 
-  void reset() {
+  void reset(bool print_on_async_exceptions = false, bool in_order = true) {
     std::lock_guard<std::mutex> lock(m_mutex);
     // The queues are shared_ptrs and the ref counts of the shared_ptrs increase
     // only in wait_and_throw(). If there is no other thread calling
     // wait_and_throw(), the queues will be destructed. The destructor waits for
     // all commands executing on the queue to complete. It isn't possible to
     // destroy a queue immediately. This is a synchronization point in SYCL.
     _queues.clear();
-    // create new default queue.
-    _queues.push_back(
-        std::make_shared<sycl::queue>(_ctx, *this, detail::exception_handler,
-                                      sycl::property::queue::in_order()));
-    _saved_queue = _default_queue = _queues.front().get();
+    // create new default queue
+    // calls create_queue_impl since we already have a locked m_mutex
+    _saved_queue = _default_queue =
```

Comment:
Not sure I really like this syntax. Feel free to keep if you feel its fine, but I would consider this unnecessarily difficult to read